### PR TITLE
Bip4 67 envconsul integration

### PR DESCRIPTION
**Pull request by sheppard-brandon-bah**
## Hey, I just made a Pull Request!

READY FOR REVIEW!!

### CODE REVIEW

**Comment by sheppard-brandon-bah** [comment]
@ThomasQuinnKelly @quijones this is finally ready to go I believe. Please give it a look aver and let me know of any questions. Thanks!

File: app-config.yaml [hunk 1]
```diff
@@ -1,9 +1,9 @@
 app:
-  title: BIH Backstage App
+  title: "Benefits Integration Hub"
```
[the remark this answers is not in the document]
Comment (bjsheppard) [reply app-config.yaml:2]: ha can never remember which one will update thanks!

File: app-config.yaml [hunk 2]
```diff
@@ -52,31 +52,31 @@ proxy:
...
-#     techdocs: local
+#     type: 'awsS3'
+#     awsS3:
+#       bucketName: ${BIH_AWS_BUCKET_NAME}
+#       region: ${BIH_AWS_BUCKET_REGION}
+#       credentials:
+#         accessKeyId: ${AWS_ACCESS_KEY_ID}
+#         secretAccessKey: ${AWS_ACCESS_KEY_SECRET}
+
+techdocs:
```
Comment (ThomasQuinnKelly) [inline app-config.yaml:66]: If this change is permanent then does this documentation need to be altered to accompany it? 

https://github.com/department-of-veterans-affairs/bih/blob/develop/README.md

File: config.hcl [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  kill_timeout = "2s"
+}
+
+kill_signal = "SIGINT"
+log_level = "warn"
+max_stale = "10m"
+
+prefix {
+  no_prefix = true
+  path = "config/bih/development"
```
Comment (ThomasQuinnKelly) [inline config.hcl:37]: Would this path or any other location need to be configurable for test / uat?

File: config.hcl [hunk 4]
```diff
@@ -0,0 +1,43 @@
+consul {
```
Comment (quijones) [inline config.hcl:1]: I dont think we need this file anymore sense we wont be building locally

File: packages/backend/Dockerfile [hunk 5]
```diff
@@ -48,4 +45,4 @@ USER node
 EXPOSE 7000
 
 # Run the app at container start.
-CMD ["node", "packages/backend", "--config", "app-config.yaml"]
\ No newline at end of file
+CMD ["envconsul", "-config", "config.hcl", "node", "packages/backend", "--config", "app-config.yaml"]
```
Comment (quijones) [inline packages/backend/Dockerfile:48]: Same here, if we are going to build locally then we no longer need to reference envconsul in the CMD directive right?

File: config.hcl (hunk 4 above)
Comment (bjsheppard) [reply config.hcl:1]: yeah we could remove it. Hat that the config is only found in K8s stuff tough. But guess that the way it be. I am indifferent.

File: packages/backend/Dockerfile (hunk 5 above)
Comment (bjsheppard) [reply packages/backend/Dockerfile:48]: same thing, just trying to be a little more transparent, put in the dockerfile what is actually getting run in k8s deployment. Maybe that is not the best approach.

File: scripts/entrypoint.sh [hunk 6]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+# app container entrypoint script for starting envconsul as
+# a sub-process then running the application
+
+/bin/envconsul -consul-addr=${CONSUL_ADDR} -consul-token=${CONSUL_TOKEN} -prefix="config/bih/development" node packages/backend --config app-config.yaml
```
[the remark this answers is not in the document]
Comment (bjsheppard) [reply scripts/entrypoint.sh:6]: entrypoint can be removed, missed that one. Will remove that thanks

File: packages/backend/Dockerfile (hunk 5 above)
Comment (bjsheppard) [reply packages/backend/Dockerfile:48]: @quijones probably going to leave this as is unless any major concerns here.

**Review by ThomasQuinnKelly — APPROVED** [review]

**Pull request merged**